src/commands/command.ts

```diff
@@ -16,17 +18,19 @@ export abstract class Command {
     abstract triggerCriteria: TriggerCriteria[];
     requiredPerms: PermissionsLevel = PermissionsLevel.USER;
 
-    abstract validate(message: Message, trigger: ConcreteTrigger, args: string[]): boolean;
-    abstract execute(message: Message, trigger: ConcreteTrigger, args: string[]): void;
+    abstract validate(message: Message, trigger: ConcreteTrigger): boolean;
+    abstract execute(message: Message, trigger: ConcreteTrigger): void;
 
-    checkTriggers(message: Message, keyword: string, args: string[]): ConcreteTrigger | null {
+    checkTriggers(message: Message): ConcreteTrigger | null {
         const trigger: Partial<ConcreteTrigger> = {
             command: this
         };
+        
+        trigger.args = message.content.split(/ +/g);
```

Comment:
I still don't see why `args: string[]` can't still be in the main message handler. It's in no way specific to string message types, and isn't really related to the trigger of a message.